tests/BUILD.go

```diff
@@ -0,0 +1,72 @@
+package tests
+
+import (
+	"libsupcxx/RULES/libsupcxx"
+	"libsupcxx/io"
+)
+
+var Test00 = libsupcxx.Binary{
+	Out:  out("test-00-global-ctors-dtors.elf"),
+	Srcs: ins("test-00-global-ctors-dtors.cc"),
+	Deps: []libsupcxx.Dep{io.Lib},
```

Comment:
I wonder whether it makes sense to always add this dependency in the `libsubcxx.Binary` rule. Will there ever be a case where one does not want to link the io lib?

Reply:
This is based on what we have for baremetal. One reason to keep it out of the default build rule is that io isn't really a stdcxx library, at least the way it's implemented here: this is also why it's included as `#include "libsupcxx/io/*"` and not as `#include <io/*>`, and IMO anything that is included this way should either be local or linked explicitly. Then again I can't see why one wouldn't want to link it, so it could be a handy default.

Reply:
I realised that io/string.hh defines some pretty crucial (and standard) C string functions, that are required by raspi3 boot code (and eventually, linux-x86_64). I'm not sure whether io is the best place to put them (libsupcxx/ probably is, since that's where stdlib functions are defined, and that's always included). But as of now, io is a dependency of boot code, which I'm not too fond of.